x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/kerberos/KerberosRealmTests.java

```diff
@@ -125,14 +128,30 @@ public void testKerberosRealmWithInvalidKeytabPathConfigurations() throws IOExce
             expectedErrorMessage = "configured service key tab file [" + keytabPath + "] does not exist";
             break;
         case "keytabPathWithNoReadPermissions":
-            filePerms = PosixFilePermissions.fromString("---------");
-            final String keytabFileName = randomAlphaOfLength(5) + ".keytab";
-            final FileAttribute<Set<PosixFilePermission>> fileAttributes = PosixFilePermissions.asFileAttribute(filePerms);
-            try (SeekableByteChannel byteChannel = Files.newByteChannel(dir.resolve(keytabFileName),
-                    EnumSet.of(StandardOpenOption.CREATE_NEW, StandardOpenOption.WRITE), fileAttributes)) {
-                byteChannel.write(ByteBuffer.wrap(randomByteArrayOfLength(10)));
+            final String fileName = randomAlphaOfLength(5);
+            final Path keytabFilePath = Files.createTempFile(dir, fileName, ".keytab");
+            Files.write(keytabFilePath, randomAlphaOfLength(5).getBytes(StandardCharsets.UTF_8));
+            final Set<String> supportedAttributes = keytabFilePath.getFileSystem().supportedFileAttributeViews();
+            if (supportedAttributes.contains("posix")) {
+                final PosixFileAttributeView fileAttributeView = Files.getFileAttributeView(keytabFilePath, PosixFileAttributeView.class);
+                fileAttributeView.setPermissions(PosixFilePermissions.fromString("---------"));
+            } else if (supportedAttributes.contains("acl")) {
+                final UserPrincipal principal = Files.getOwner(keytabFilePath);
+                final AclFileAttributeView view = Files.getFileAttributeView(keytabFilePath, AclFileAttributeView.class);
+                final AclEntry entry = AclEntry.newBuilder()
+                        .setType(AclEntryType.DENY)
+                        .setPrincipal(principal)
+                        .setPermissions(AclEntryPermission.READ_DATA, AclEntryPermission.READ_ATTRIBUTES).build();
+                final List<AclEntry> acl = view.getAcl();
+                acl.add(0, entry);
+                view.setAcl(acl);
+            } else if (supportedAttributes.contains("dos")) {
+                final DosFileAttributeView fileAttributeView = Files.getFileAttributeView(keytabFilePath, DosFileAttributeView.class);
+                fileAttributeView.setReadOnly(false);
```

Comment:
? How does making it not-read only help?

Reply:
The case statement is `keytabPathWithNoReadPermissions` but read-only false is not no-read, it's can-write.

So the setup for DOS isn't consistent with the test scenario.

Reply:
Yes, I would ditch DOS (as you have done).
I don't think we run any CI on non-ACL aware platforms. If it turns out we do, then we should just do an `assumeFalse` as the test is not possible on that platform.